Error when `docker buld .` on Mac m1 pro

### Environment

relay@22.7.0
docker@latest
M1 pro macbook pro.

### Steps to Reproduce

```
docker build .
[+] Building 211.3s (19/24)                                                                                                                                                                                                                                               
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                 0.0s
 => => transferring dockerfile: 2.50kB                                                                                                                                                                                                                               0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                    0.0s
 => => transferring context: 35B                                                                                                                                                                                                                                     0.0s
 => [internal] load metadata for docker.io/library/debian:buster-slim                                                                                                                                                                                                2.2s
 => [internal] load metadata for docker.io/getsentry/sentry-cli:1                                                                                                                                                                                                    2.3s
 => [internal] load metadata for docker.io/amd64/centos:7                                                                                                                                                                                                            2.2s
 => [internal] load build context                                                                                                                                                                                                                                    0.2s
 => => transferring context: 366.38kB                                                                                                                                                                                                                                0.2s
 => [relay-deps 1/5] FROM docker.io/amd64/centos:7@sha256:dead07b4d8ed7e29e98de0f4504d87e8880d4347859d839686a31da35a3b532f                                                                                                                                           0.0s
 => [sentry-cli 1/1] FROM docker.io/getsentry/sentry-cli:1@sha256:32bd861e849a54567506c77a0995f94c7915ad04e91989e215f4d286a5a7fbf3                                                                                                                                   0.0s
 => [stage-3 1/8] FROM docker.io/library/debian:buster-slim@sha256:702bc27ef4be73f9c1d73c1a2bc58987c59b6fc8e8e04f46c166849817ce95dc                                                                                                                                  0.0s
 => CACHED [stage-3 2/8] RUN apt-get update     && apt-get install -y ca-certificates gosu curl --no-install-recommends     && apt-get clean     && rm -rf /var/lib/apt/lists/*                                                                                      0.0s
 => CACHED [stage-3 3/8] RUN groupadd --system relay --gid 10001     && useradd --system --gid relay --uid 10001 relay                                                                                                                                               0.0s
 => CACHED [stage-3 4/8] RUN mkdir /work /etc/relay     && chown relay:relay /work /etc/relay                                                                                                                                                                        0.0s
 => CACHED [stage-3 5/8] WORKDIR /work                                                                                                                                                                                                                               0.0s
 => CACHED [relay-deps 2/5] RUN yum -y update     && yum -y install centos-release-scl epel-release     && yum -y install cmake3 devtoolset-10 git     llvm-toolset-7.0-clang-devel     && yum clean all     && rm -rf /var/cache/yum     && ln -s /usr/bin/cmake3   0.0s
 => CACHED [relay-deps 3/5] RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --profile minimal                                                                                                                                            0.0s
 => CACHED [relay-deps 4/5] COPY --from=sentry-cli /bin/sentry-cli /bin/sentry-cli                                                                                                                                                                                   0.0s
 => CACHED [relay-deps 5/5] WORKDIR /work                                                                                                                                                                                                                            0.0s
 => CACHED [relay-builder 1/4] COPY . .                                                                                                                                                                                                                              0.0s
 => ERROR [relay-builder 2/4] RUN scl enable devtoolset-10 llvm-toolset-7.0 --     make build-linux-release     TARGET=x86_64-unknown-linux-gnu     RELAY_FEATURES=ssl,processing,crash-handler                                                                    208.7s
------                                                                                                                                                                                                                                                                    
 > [relay-builder 2/4] RUN scl enable devtoolset-10 llvm-toolset-7.0 --     make build-linux-release     TARGET=x86_64-unknown-linux-gnu     RELAY_FEATURES=ssl,processing,crash-handler:                                                                                 
#19 6.426 cd relay && cargo build --release --locked --features ssl,processing,crash-handler --target=x86_64-unknown-linux-gnu
#19 9.174     Updating crates.io index
#19 208.6 /bin/bash: line 1:  1214 Killed                  cargo build --release --locked --features ssl,processing,crash-handler --target=x86_64-unknown-linux-gnu
#19 208.6 make: *** [Makefile:27: build-linux-release] Error 137
------
executor failed running [/bin/sh -c scl enable devtoolset-10 llvm-toolset-7.0 --     make build-linux-release     TARGET=${BUILD_TARGET}     RELAY_FEATURES=${RELAY_FEATURES}]: exit code: 2
```
### Expected Result

What you thought would happen.

### Actual Result

What actually happened. Maybe a screenshot/recording? Maybe some logs?

@jjbayer  relay latest version can't build on m1 pro mac too.
```
relay git:(master) docker build -t test-relay .
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
[+] Building 1421.1s (20/25)                                                                                                                                                                                                                                       
 => [internal] booting buildkit                                                                                                                                                                                                                               0.5s
 => => starting container buildx_buildkit_mybuilder0                                                                                                                                                                                                          0.5s
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                          0.0s
 => => transferring dockerfile: 2.67kB                                                                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                             0.0s
 => => transferring context: 348B                                                                                                                                                                                                                             0.0s
 => [internal] load metadata for docker.io/library/debian:buster-slim                                                                                                                                                                                         3.0s
 => [internal] load metadata for docker.io/getsentry/sentry-cli:1                                                                                                                                                                                             2.2s
 => [internal] load metadata for docker.io/amd64/centos:7                                                                                                                                                                                                     2.2s
 => CACHED [relay-deps 1/5] FROM docker.io/amd64/centos:7@sha256:dead07b4d8ed7e29e98de0f4504d87e8880d4347859d839686a31da35a3b532f                                                                                                                             0.0s
 => => resolve docker.io/amd64/centos:7@sha256:dead07b4d8ed7e29e98de0f4504d87e8880d4347859d839686a31da35a3b532f                                                                                                                                               0.0s
 => [internal] load build context                                                                                                                                                                                                                             0.2s
 => => transferring context: 360.30kB                                                                                                                                                                                                                         0.2s
 => CACHED [sentry-cli 1/1] FROM docker.io/getsentry/sentry-cli:1@sha256:32bd861e849a54567506c77a0995f94c7915ad04e91989e215f4d286a5a7fbf3                                                                                                                     0.0s
 => => resolve docker.io/getsentry/sentry-cli:1@sha256:32bd861e849a54567506c77a0995f94c7915ad04e91989e215f4d286a5a7fbf3                                                                                                                                       0.0s
 => [stage-3 1/8] FROM docker.io/library/debian:buster-slim@sha256:702bc27ef4be73f9c1d73c1a2bc58987c59b6fc8e8e04f46c166849817ce95dc                                                                                                                           0.0s
 => => resolve docker.io/library/debian:buster-slim@sha256:702bc27ef4be73f9c1d73c1a2bc58987c59b6fc8e8e04f46c166849817ce95dc                                                                                                                                   0.0s
 => [relay-deps 2/5] RUN yum -y update     && yum -y install centos-release-scl epel-release     && yum -y install cmake3 devtoolset-10 git     llvm-toolset-7.0-clang-devel     && yum clean all     && rm -rf /var/cache/yum     && ln -s /usr/bin/cmake  928.2s
 => CACHED [stage-3 2/8] RUN apt-get update     && apt-get install -y ca-certificates gosu curl --no-install-recommends     && apt-get clean     && rm -rf /var/lib/apt/lists/*                                                                               0.0s
 => CACHED [stage-3 3/8] RUN groupadd --system relay --gid 10001     && useradd --system --gid relay --uid 10001 relay                                                                                                                                        0.0s
 => CACHED [stage-3 4/8] RUN mkdir /work /etc/relay     && chown relay:relay /work /etc/relay                                                                                                                                                                 0.0s
 => CACHED [stage-3 5/8] WORKDIR /work                                                                                                                                                                                                                        0.0s
 => [relay-deps 3/5] RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs     | sh -s -- -y --profile minimal --default-toolchain=1.65.0                                                                                                           175.9s
 => [relay-deps 4/5] COPY --from=sentry-cli /bin/sentry-cli /bin/sentry-cli                                                                                                                                                                                   0.0s 
 => [relay-deps 5/5] WORKDIR /work                                                                                                                                                                                                                            0.0s 
 => [relay-builder 1/4] COPY . .                                                                                                                                                                                                                              0.3s 
 => ERROR [relay-builder 2/4] RUN scl enable devtoolset-10 llvm-toolset-7.0 --     make build-linux-release     TARGET=x86_64-unknown-linux-gnu     RELAY_FEATURES=ssl,processing,crash-handler                                                             312.9s 
------                                                                                                                                                                                                                                                             
 > [relay-builder 2/4] RUN scl enable devtoolset-10 llvm-toolset-7.0 --     make build-linux-release     TARGET=x86_64-unknown-linux-gnu     RELAY_FEATURES=ssl,processing,crash-handler:                                                                          
#0 6.094 cd relay && cargo build --release --locked --features ssl,processing,crash-handler --target=x86_64-unknown-linux-gnu                                                                                                                                      
#0 8.642     Updating crates.io index                                                                                                                                                                                                                              
#0 312.7 /bin/bash: line 1:  1215 Killed                  cargo build --release --locked --features ssl,processing,crash-handler --target=x86_64-unknown-linux-gnu
#0 312.7 make: *** [Makefile:27: build-linux-release] Error 137
------
Dockerfile:54
--------------------
  53 |     # Build with the modern compiler toolchain enabled
  54 | >>> RUN scl enable devtoolset-10 llvm-toolset-7.0 -- \
  55 | >>>     make build-linux-release \
  56 | >>>     TARGET=${BUILD_TARGET} \
  57 | >>>     RELAY_FEATURES=${RELAY_FEATURES}
  58 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c scl enable devtoolset-10 llvm-toolset-7.0 --     make build-linux-release     TARGET=${BUILD_TARGET}     RELAY_FEATURES=${RELAY_FEATURES}" did not complete successfully: exit code: 2
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -51,7 +51,8 @@ ENV RELAY_FEATURES=${RELAY_FEATURES}
 COPY . .
 
 # Build with the modern compiler toolchain enabled
-RUN scl enable devtoolset-10 llvm-toolset-7.0 -- \
+RUN echo -e "[net]\ngit-fetch-with-cli = true" > $CARGO_HOME/config \
+    && scl enable devtoolset-10 llvm-toolset-7.0 -- \
     make build-linux-release \
     TARGET=${BUILD_TARGET} \
     RELAY_FEATURES=${RELAY_FEATURES}

